### ISSUE TITLE
Implement new Concept Exercise: lists

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/java#1862]
This issue describes how to implement the `lists` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Lists in Java. We'll use the list type to teach the student about generics too. We should be using the list with more than one data type to show that is possible. 

The `list` collection type was chosen as the first generic type for the following reasons:

- It is the most commonly used generic type.
- They are a common data type in many language.

## Learning objectives

- Know what generic types are.
- Know of the existence of the `List<T>` type.
- Know how a list is different from an array.
- Know how to define a list.
- Know how to add and remove elements from a list.
- Know how to access elements in a list by index.
- Know how to iterate over elements in a list.
- Know some basic list functions (like finding the index of an element in a list or sorting a list).

## Out of scope

- Generic functions.
- Generic constraints.
- Memory and performance characteristics.
- Concurrency issues.
- Co-/contravariance.
- Equality.
- List resizing due to it using an array as its data type.

## Concepts

This Concepts Exercise's Concepts are:

- `lists`: know of the existence of the `List<T>` type; know how a list is different from an array; know how to define a list; know how to add and remove elements from a list; know how to access elements in a list by index; know how to iterate over elements in a list; know some basic list functions (like finding the index of an element in a list).
- `generic-types`: know what generic types are.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `for-loops`: know how to use a `for-loop` to iterate over a collection.
- `arrays`: know of the `array` collection type and that it has a fixed length.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer).

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer).

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by Priya103** [comment]
If still unassigned I would like to contribute.

**Comment by mirkoperillo** [comment]
yes, thank you.
I assign it to you @Priya103 

For every doubt or suggestion feel free to ask in this issue, I'll try to help you

**Comment by mirkoperillo** [comment]
@Priya103  I have merged the v3 stuff on the repo right know.
You can take the last updates of the main branch to start your work.

**Comment by Priya103** [comment]
Thanks @mirkoperillo

**Comment by jmrunkle** [comment]
FYI: this is blocking the `remote-control-competition` concept exercise (see https://github.com/exercism/java/issues/1992)

**Comment by jmrunkle** [comment]
With no one else investing in this one and getting two concepts for the price of one, I think I will take a stab at this.
